src/java.base/share/classes/jdk/internal/loader/URLClassPath.java

```diff
@@ -940,7 +946,7 @@ Resource getResource(final String name, boolean check) {
             if (entry != null)
                 return checkResource(name, check, entry);
 
-            if (index == null)
+            if (index == null || !ENABLE_JAR_INDEX)
```

Comment:
Is this needed? When ENABLE_JAR_INDEX is false then I assume index will always be null. I'm only asking is that it would be nice if ENABLE_JAR_INDEX was checked in one place rather than two.

Reply:
Yes, the check is redundant, and I removed it.